From 81d0fdaf2670c4f9bf70d9ae3681cf9401c82082 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Thu, 4 Apr 2013 00:14:09 +0200
Subject: [PATCH 0997/1083] drm/rcar-du: Take offsets into account in the
 mode_set_base handler

Store the x and y offsets passed to the mode_set_base handler in the
plane structure. The offsets will be programmed to the hardware when
updating the plane base address.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit e308401dfca467f107b4f8519196845b4b4105a4)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit 5314afb9293ab03edcee9a66daf9f0143f634a6d)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 53dbe36..c66fa4c 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -445,6 +445,11 @@ static void rcar_du_crtc_mode_commit(struct drm_crtc *crtc)
 static int rcar_du_crtc_mode_set_base(struct drm_crtc *crtc, int x, int y,
 				      struct drm_framebuffer *old_fb)
 {
+	struct rcar_du_crtc *rcrtc = to_rcar_crtc(crtc);
+
+	rcrtc->plane->src_x = x;
+	rcrtc->plane->src_y = y;
+
 	rcar_du_crtc_update_base(to_rcar_crtc(crtc));
 
 	return 0;
-- 
1.8.3.2

